pkg/clients/s3/bucket.go

```diff
@@ -0,0 +1,118 @@
+	DeleteBucketReplicationRequest(input *s3.DeleteBucketReplicationInput) s3.DeleteBucketReplicationRequest
+
+	PutBucketRequestPaymentRequest(input *s3.PutBucketRequestPaymentInput) s3.PutBucketRequestPaymentRequest
+	GetBucketRequestPaymentRequest(input *s3.GetBucketRequestPaymentInput) s3.GetBucketRequestPaymentRequest
+
+	PutBucketTaggingRequest(input *s3.PutBucketTaggingInput) s3.PutBucketTaggingRequest
+	GetBucketTaggingRequest(input *s3.GetBucketTaggingInput) s3.GetBucketTaggingRequest
+	DeleteBucketTaggingRequest(input *s3.DeleteBucketTaggingInput) s3.DeleteBucketTaggingRequest
+
+	PutBucketAnalyticsConfigurationRequest(input *s3.PutBucketAnalyticsConfigurationInput) s3.PutBucketAnalyticsConfigurationRequest
+	GetBucketAnalyticsConfigurationRequest(input *s3.GetBucketAnalyticsConfigurationInput) s3.GetBucketAnalyticsConfigurationRequest
+
+	PutBucketLifecycleConfigurationRequest(input *s3.PutBucketLifecycleConfigurationInput) s3.PutBucketLifecycleConfigurationRequest
+	GetBucketLifecycleConfigurationRequest(input *s3.GetBucketLifecycleConfigurationInput) s3.GetBucketLifecycleConfigurationRequest
+	DeleteBucketLifecycleRequest(input *s3.DeleteBucketLifecycleInput) s3.DeleteBucketLifecycleRequest
+
+	PutBucketNotificationConfigurationRequest(input *s3.PutBucketNotificationConfigurationInput) s3.PutBucketNotificationConfigurationRequest
+	GetBucketNotificationConfigurationRequest(input *s3.GetBucketNotificationConfigurationInput) s3.GetBucketNotificationConfigurationRequest
+
+	GetBucketAclRequest(*s3.GetBucketAclInput) s3.GetBucketAclRequest
+	PutBucketAclRequest(*s3.PutBucketAclInput) s3.PutBucketAclRequest
+}
+
+// NewClient returns a new client using AWS credentials as JSON encoded data.
+func NewClient(ctx context.Context, credentials []byte, region string, auth awsclients.AuthMethod) (BucketClient, error) {
+	cfg, err := auth(ctx, credentials, awsclients.DefaultSection, region)
+	if cfg == nil {
+		return nil, err
+	}
+	return s3.New(*cfg), nil
+}
+
+// IsNotFound helper function to test for ErrCodeNoSuchEntityException error
+func IsNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	if bucketErr, ok := err.(awserr.Error); ok && bucketErr.Code() == "NotFound" {
+		return true
+	}
+	return false
+}
+
+// GenerateCreateBucketInput creates the input for CreateBucket S3 Client request
+func GenerateCreateBucketInput(name string, s v1beta1.BucketParameters) *s3.CreateBucketInput {
+	cbi := &s3.CreateBucketInput{
+		ACL:                        s3.BucketCannedACL(aws.StringValue(s.ACL)),
+		Bucket:                     aws.String(name),
+		GrantFullControl:           s.GrantFullControl,
+		GrantRead:                  s.GrantRead,
+		GrantReadACP:               s.GrantReadACP,
+		GrantWrite:                 s.GrantWrite,
+		GrantWriteACP:              s.GrantWriteACP,
+		ObjectLockEnabledForBucket: s.ObjectLockEnabledForBucket,
+	}
+	if awsclients.StringValue(s.LocationConstraint) != "us-east-1" {
```

Comment:
Why do we need to do this check?

Reply:
If we try to pass in "us-east-1", the SDK throws an InvalidLocationConstraint error. It seems like the SDK expects an empty string if you want the bucket to exist on "us-east-1".